spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SnapshotTableSparkAction.java

```diff
@@ -124,10 +124,12 @@ private SnapshotTable.Result doExecute() {
     StagedSparkTable stagedTable = stageDestTable();
     Table icebergTable = stagedTable.table();
 
-    // TODO: Check the dest table location does not overlap with the source table location
-
     boolean threw = true;
     try {
+      Preconditions.checkArgument(
+          !sourceTableLocation().equals(icebergTable.location()),
+          "The destination table location overlaps with the source table location.");
```

Comment:
The following `.` can be removed. Most existing error messages don't end with `.` unless they have several sentences.

Reply:
If you decide to remove the full stop, don't forget to remove it from the test as well so that it will not fail.

Reply:
Thank you for helping review the code! I have made improvements to this part of the code.